raiden_contracts/tests/test_channel_close.py

```diff
@@ -181,6 +181,56 @@ def test_close_wrong_sender(
         ).transact({'from': C})
 
 
+def test_close_nonce_zero(
+        get_accounts,
+        token_network,
+        create_channel,
+        create_balance_proof,
+):
+    (A, B) = get_accounts(2)
+    vals_B = ChannelValues(
+        deposit=20,
+        transferred=5,
+        locksroot=fake_bytes(32, '03'),
+        claimable_locked=3,
+        unclaimable_locked=4,
+        nonce=0,
+    )
+    # Create channel and deposit
+    channel_identifier = create_channel(A, B)[0]
+
+    # Create balance proofs
+    balance_proof_B = create_balance_proof(
+        channel_identifier,
+        B,
+        vals_B.transferred,
+        vals_B.locked,
+        vals_B.nonce,
+        vals_B.locksroot,
+    )
+
+    token_network.functions.closeChannel(
+        channel_identifier,
+        B,
+        *balance_proof_B,
+    ).transact({'from': A})
+
+    # Even though we somehow provide valid values for the balance proof, they are not taken into
```

Comment:
We added a `if (nonce > 0 ) then update balance proof` to avoid wrongly constructed balance proofs to be stored on-chain & lower gas cost. Added a test to check this behavior.